src/radical/entk/appman/wfprocessor.py

```diff
@@ -249,328 +287,235 @@ def _enqueue(self, local_prof):
             self._logger.info('Enqueue thread terminated')
             mq_connection.close()
 
-            local_prof.prof('terminating enqueue-thread', uid=self._uid)
+            self._prof.prof('terminating enqueue-thread', uid=self._uid)
 
         except KeyboardInterrupt:
 
-            self._logger.exception('Execution interrupted by user (you probably hit Ctrl+C), ' +
-                               'trying to cancel enqueuer thread gracefully...')
-
+            self._logger.exception('Execution interrupted by user (you \
+                                    probably hit Ctrl+C), trying to cancel \
+                                    enqueuer thread gracefully...')
             mq_connection.close()
-
             raise KeyboardInterrupt
 
         except Exception, ex:
 
-            self._logger.exception('Error in enqueue-thread: %s' % ex)
+            self._logger.exception('Error in enqueue-thread')
             try:
                 mq_connection.close()
             except Exception as ex:
                 self._logger.warning('mq_connection not created, %s' % ex)
 
             raise
 
```

Comment:
If you catch an exception, log it as exception, and raise it again, then the same exception and the same trace get logged twice.

Reply:
logging the exception doesn't the parent object that an exception was encountered though right? I raise it again so that parent object essentially stops and does not try to continue. Maybe there is another/better way to do that?

Reply:
I am sorry, I don't understand your comment :(

> logging the exception doesn't the parent object that an exception was encountered though right?

Is there a verb missing?  What is expected to (not) happen to the parent object?  What do you mean with parent object - parent of what?

>  I raise it again so that parent object essentially stops and does not try to continue. 

An object does not continue - do you mean this method?  That will sure be aborted on `raise` - but it will also be aborted if you let the exception fall through.

Reply:
Uhmm, I don't understand myself either :-| Think I missed several words, let me try again!

Function X calls function Y. Function Y raises an exception which is logged. But upon that exception, I want function X to stop executing as well, so I raise it again.

```python
def func_x():
    func_y()
    print 'not to print'

def func_y():
    try:
        x = 1/0
    except:
        print 'bad math'

>>> func_x()
bad math
not to print
```

vs

```python
def func_x():
    func_y()
    print 'not to print'

def func_y():
    try:
        x = 1/0
    except:
        print 'bad math'
        raise

>>> func_x()
bad math
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in func_x
  File "<stdin>", line 3, in func_y
ZeroDivisionError: integer division or modulo by zero
```

Does that help? If you meant, we shouldn't be logging again in function X (although logging is not there in my example), I may agree, but I need to check.

Reply:
It does.  My suggestion is to just not except it:
```
>>> def func_x():
...     func_y()
...     print 'not to print'
...
>>> def func_y():
...     x = 1/0
...
>>> func_x()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in func_x
  File "<stdin>", line 2, in func_y
ZeroDivisionError: integer division or modulo by zero
```

Whover calls `func_y()` / `_execute_post_exec()` needs to be aware that it can raise an exception, needs to except that, and will lock the **same** stacktrace again.  I don't see the benefit of the double try/catch?